aten/src/ATen/native/mps/operations/TensorCompare.mm

```diff
@@ -146,17 +146,30 @@ void clamp_tensor_out_mps(const Tensor& input_t,
       cachedGraph = static_cast<CachedGraph*>(tmpCachedGraph);
     }
 
-    auto inputPlaceholder = Placeholder(cachedGraph->inputTensor, input_t);
-    auto outputPlaceholder = Placeholder(cachedGraph->outputTensor, output_t);
+    bool gatherTensorData = true;
+    if (!output_t.is_contiguous() || output_t.is_view()) {
```

Comment:
This needs to be centrally fixed, rather than just for Compare operations. Here the fix is to allocate the output using empty restrided and then pass through the Strided tensor. The major concern I have is the performance implications, because we will end up doing Gather for strided inputs which in this case can be avoided if output is also strided.

Let us propose a PR which fixes it more centrally.

Reply:
Actually we have been using the same fix for a while: #94260. I understood that this is not the best way to fix, but my aim was to unblock some users who have been affected by this bug for months, despite some performance implications.

Reply:
@kulinseth Hi, would like to know any update regarding this?

Reply:
If this solution is not performant enough, could we at least throw an error when we know the result will be wrong? This kind of silent incorrectness makes the mps backend nearly unusable.

Reply:
Hi @qqaatw, I agree, lets unblock users and we can centrally fix it later.